catalog/app/utils/perspective.ts

```diff
@@ -39,40 +41,43 @@ function usePerspective(
   attrs: React.HTMLAttributes<HTMLDivElement>,
   settings?: boolean,
 ) {
-  const [tableData, setTableData] = React.useState<TableData | null>(null)
+  const table = React.useRef<Table | null>(null)
+  const viewer = React.useRef<HTMLPerspectiveViewerElement | null>(null)
 
-  React.useEffect(() => {
-    let table: Table, viewer: HTMLPerspectiveViewerElement
+  const [state, setState] = React.useState<State>({
+    toggleConfig: () => viewer.current?.toggleConfig(),
```

Comment:
i don't see how this is a part of the state. why not just make it a memoized function of `viewer`?

Reply:
I didn't understand this
You don't like `state` name? Or, do you mean I can pass the `viewer`, not just function?

Reply:
since you had `viewer` in scope (in the ref-based implementation), you could've made `toggleConfig` a memoized function of viewer, e.g. `const toggleConfig = React.useCallback(() => viewer.current?.toggleConfig(), [viewer])`, but doesn't seem relevant anymore